### ISSUE TITLE
ResolverFully does not resolve $ref in ComposedSchema with properties

### ISSUE THREAD

**Issue by scottr-ad** [swagger-api/swagger-parser#1161]
When resolving fully a schema with anyOf, allOf and/or oneOf and object properties with references, as in the example schema below, the property references are missed and remain as $ref.

swagger.yaml
---------------
```
openapi: "3.0.2"
info:
  version: 15.3.0
  title: test
paths:
  /add-pets:
    put:
      requestBody:
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/Pets'
      responses:
        '200':
          description: Expected response to a valid request
          content:
            text/plain:
              schema:
                type: string
components:
  schemas:
    Pets:
      type: object
      oneOf:
        - required: [petType, colouring]
        - required: [breed, colouring]
      properties:
        petType:
          type: string
        breed:
          $ref: 'common.yaml#/components/schemas/Breed'
        colouring:
          $ref: 'common.yaml#/components/schemas/Colouring'
```

common.yaml
----------------
```
openapi: "3.0.0"
info:
  version: 15.3.0
  title: "Common Data Types"
paths: {}
components:
  schemas:
    Breed:
      type: object
      properties:
        name:
          type: string
        family:
          type: string
    Colouring:
      type: object
      properties:
        primary:
          $ref: '#/components/schemas/Colour'
        secondary:
          $ref: '#/components/schemas/Colour'
    Colour:
      type: string
      enum: ['black', 'white', 'tan', 'red', 'blue']
```

This is because in ResolverFully.resolveSchema, after the anyOf, allOf and oneOf parts of ComposedSchema are resolved, the method returns immediately, without also resolving the object's properties. 

A fix for this would be to not return immediately after resolving the anyOf, allOf and oneOf parts of ComposedSchema, allowing it to proceed to the next code block, where properties are processed.

**Comment by scottr-ad** [comment]
Pull request #1162 raised to remedy this issue. Could one of the regular contributors please review this PR and either approve it or provide feedback?

**Comment by gracekarina** [comment]
Hi, did you try the option resolveCombinators of parser?

**Comment by scottr-ad** [comment]
Hi @gracekarina  

I'm using `https://bitbucket.org/atlassian/swagger-request-validator` to validate messages against the schema, and it does not set that option. This is to preserve the semantics of oneOf, allOf and anyOf for validation.

However, looking through the ResolverFully code, even if the resolveCombinators option were set, it would only create a new aggregated schema if:
(a) allOf is used, or
(b) more than 1 of allOf, anyOf or oneOf is used (ie. 'adjacent' is true).

Neither of which would apply to the example schema above.

**Comment by gracekarina** [comment]
ok then I'll review the case and ket you know tomorrow.

**Comment by gracekarina** [comment]
closed by #1162
